src/app/resources/view-resources/resources-view.component.ts

```diff
@@ -89,6 +102,66 @@ export class ResourcesViewComponent implements OnInit, OnDestroy {
       this.pdfSrc = this.sanitizer.bypassSecurityTrustResourceUrl(this.resourceSrc);
     }
     this.couchSrc = this.urlPrefix + resource._id + '/' + filename;
+    // resource rating
+    this.couchService.get('ratings/_all_docs?include_docs=true')
+      .subscribe((data) => {
+        this.ratings = data.rows.map(ratings => {
+          return ratings.doc;
+        }).filter(rt  => {
+          return rt['type'] === 'resource' && rt['item'] === resource._id;
+        });
+        let rate_sum = 0;
+        let has_rated = 0;
+        let total_rating = 0;
+        let male_rating = 0;
+        let female_rating = 0;
+        this.ratings.map(rate => {
+          has_rated = (rate.user === this.userService.get().name) ? rate.rate : has_rated;
+          total_rating++;
+          (rate.gender === 'M') ? male_rating++ : female_rating++ ;
+          rate_sum = rate_sum + parseInt(rate.rate, 10);
+        });
+        this.resource.rating = rate_sum;
+        this.resource.has_rated = has_rated;
+        this.resource.female_rating = (female_rating / total_rating) * 100;
+        this.resource.male_rating = (male_rating / total_rating) * 100;
+        this.resource.total_rating = total_rating;
+      }, (error) => console.log(error));
+  }
+
+  rate(resource_id) {
+    const title = 'Rating';
+    const type = 'rating';
+    // need to show star rating insted of typebox
+    const fields =
+      [
+        { 'label': 'Rate', 'type': 'radio', 'name': 'rate', 'placeholder': 'Your Rating', 'required': false },
+        { 'label': 'Comment', 'type': 'textarea', 'name': 'comment', 'placeholder': 'Leave your comment', 'required': false }
+      ];
+    const validation = {
+      rate: [ '' ],
+      comment: [ '' ]
+    };
+    this.dialogsFormService
+      .confirm(title, type, fields, validation, '')
+      .debug('Dialog confirm')
+      .subscribe((res) => {
+        if (res !== undefined) {
+          const datas = {
+            'user': this.userService.get().name,

```

Comment:
We should insert the full user object here.  Adding a `getProfile` method to the `userService` which fetches that information would work.

Reply:
That means getProfile need to fetch his/her data from _users document , so Should I query it when user is set in auth-guard.service.ts.
After 
this.userService.set(res.userCtx);

Reply:
Yes, do a get request for the full document from the _users database.  The auth-guard service runs on every route change, so if we run it from there we should only request profile data if the user name has changed.